lib/tasks/locale.rake

```diff
@@ -209,21 +195,24 @@ namespace :locale do
     ]
 
     tmp_dir = Rails.root.join('locale', 'tmp').to_s
-    Dir.mkdir(tmp_dir, 0o700)
+    FileUtils.rm_rf(tmp_dir)
+    FileUtils.mkdir(tmp_dir)
     extra_pots.each do |url|
       pot_file = "#{tmp_dir}/#{url.split('/')[-1]}"
       ManageIQ::Environment.system!('curl', '-f', '-o', pot_file, url)
       pot_files << pot_file
     end
 
-    system('rmsgcat', '--sort-by-msgid', '-o', Rails.root.join('locale', 'manageiq-all.pot').to_s, Rails.root.join('locale', 'manageiq.pot').to_s, *pot_files)
-    system('mv', '-v', Rails.root.join('locale', 'manageiq-all.pot').to_s, Rails.root.join('locale', 'manageiq.pot').to_s)
-    system('rmsgmerge', '--sort-by-msgid', '--no-fuzzy-matching', '-o', Rails.root.join('locale', 'en', 'manageiq-all.po').to_s, Rails.root.join('locale', 'en', 'manageiq.po').to_s, Rails.root.join('locale', 'manageiq.pot').to_s)
-    system('mv', '-v', Rails.root.join('locale', 'en', 'manageiq-all.po').to_s, Rails.root.join('locale', 'en', 'manageiq.po').to_s)
-    system('rm', '-rf', tmp_dir)
+    # Generate the file locations for reference.
+    # msgcat will concatenate and retain all file location information, whereas rmsgcat will only retain file locations from the last file ARG
+    system('msgcat', '--sort-output', '-o', Rails.root.join('locale', 'manageiq-with-file-locations.pot').to_s, Rails.root.join('locale', 'manageiq.pot').to_s, *pot_files)
+
+    system('rmsgcat', '--sort-by-msgid', '--no-location', '-o', Rails.root.join('locale', 'tmp', 'manageiq-all.pot').to_s, Rails.root.join('locale', 'manageiq.pot').to_s, *pot_files)
+    system('mv', '-v', Rails.root.join('locale', 'tmp', 'manageiq-all.pot').to_s, Rails.root.join('locale', 'manageiq.pot').to_s)
+    system('rmsgmerge', '--sort-by-msgid', '--no-location', '--no-fuzzy-matching', '-o', Rails.root.join('locale', 'tmp', 'manageiq-all.po').to_s, Rails.root.join('locale', 'en', 'manageiq.po').to_s, Rails.root.join('locale', 'manageiq.pot').to_s)
+    system('mv', '-v', Rails.root.join('locale', 'tmp', 'manageiq-all.po').to_s, Rails.root.join('locale', 'en', 'manageiq.po').to_s)
```

Comment:
Minor refactor to make this easier to read?

```suggestion
    locale_dir = Rails.root.join("locale")
    locale_tmp_dir = locale_dir.join("tmp)

    # Generate the file locations for reference.
    # msgcat will concatenate and retain all file location information, whereas rmsgcat will only retain file locations from the last file ARG
    system('msgcat', '--sort-output', '-o', locale_dir.join('manageiq-with-file-locations.pot').to_s, locale_dir.join('manageiq.pot').to_s, *pot_files)

    system('rmsgcat', '--sort-by-msgid', '--no-location', '-o', locale_tmp_dir.join('manageiq-all.pot').to_s, locale_dir.join('manageiq.pot').to_s, *pot_files)
    system('mv', '-v', locale_tmp_dir.join('manageiq-all.pot').to_s, locale_dir.join('manageiq.pot').to_s)
    system('rmsgmerge', '--sort-by-msgid', '--no-location', '--no-fuzzy-matching', '-o', locale_tmp_dir.join('manageiq-all.po').to_s, locale_dir.join('en', 'manageiq.po').to_s, locale_dir.join('manageiq.pot').to_s)
    system('mv', '-v', locale_tmp_dir.join('manageiq-all.po').to_s, locale_dir.join('en', 'manageiq.po').to_s)
```

Reply:
done